payroll/urls.py

```diff
@@ -33,4 +33,14 @@
         views.add_vacancy_page,
         name="add_vacancy",
     ),
+    path(
+        "edit/<str:cost_centre_code>/<int:financial_year>/vacancies/edit/<int:vacancy_id>",
```

Comment:
what do you think to the action on the end? feels more RESTful to me

```suggestion
        "edit/<str:cost_centre_code>/<int:financial_year>/vacancies/<int:vacancy_id>/edit",
```